pandas/tests/indexes/test_multi.py

```diff
@@ -766,6 +766,14 @@ def test_from_arrays_empty(self):
                 ValueError, "Must pass non-zero number of levels/labels"):
             MultiIndex.from_arrays(arrays=[])
 
+        # 0 levels, names defined
+        result = MultiIndex.from_arrays(arrays=[],
```

Comment:
Right, which would keep the current behavior for `.from_arrays`. I'd argue that if `.from_tuples(tuples=(), names=names)` is valid then `.from_array(arrays=[], names=names)` should also be allowed (and produce an identical object).